Geometry/EcalAlgo/src/EcalEndcapGeometry.cc

```diff
@@ -516,10 +516,10 @@ EcalEndcapGeometry::avgAbsZFrontFaceCenter() const
    return m_avgZ;
 }
 
-const CaloCellGeometry* 
-EcalEndcapGeometry::cellGeomPtr( uint32_t index ) const
+std::shared_ptr<CaloCellGeometry>  
+EcalEndcapGeometry::cellGeomPtr( uint32_t index ) 
 {
-   const CaloCellGeometry* cell ( &m_cellVec[ index ] ) ;
-   return ( m_cellVec.size() < index ||
-	    nullptr == cell->param() ? nullptr : cell ) ;
+  auto cell = std::shared_ptr<CaloCellGeometry>(&m_cellVec[index]) ;
```

Comment:
I think the problem is (for example) here, where you are wrapping `&m_cellVec[index]` in a `shared_ptr`.

You should avoid wrapping in a `shared_ptr` any pointer than it does not own, because when the `shared_ptr` goes out of scope it will try to delete it.

If you really *must* do it, you can try to prevent the deletion of these objects with something like
```
auto do_not_delete = [](CaloCellGeometry*){};
...
auto cell = std::shared_ptr<CaloCellGeometry>(&m_cellVec[index], do_not_delete) ;
```

However, I would advice against using a `shared_ptr` this way, as you will be allocating and deallocating plenty of small memory chunks during any `for` loops iterating over the cells.